src/main/java/org/wso2/lsp4intellij/utils/DocumentUtils.java

```diff
@@ -109,55 +128,51 @@ public static Position offsetToLSPPos(Editor editor, int offset) {
+            int line = max(0, Math.min(pos.getLine(), doc.getLineCount()-1));
+            String lineText = doc.getText(DocumentUtil.getLineTextRange(doc, line));
+
+            final int endCharInLine = max(0, min(lineText.length(), pos.getCharacter()));
+            String lineTextForPosition = endCharInLine > 0 ? lineText.substring(0, endCharInLine) : "";
+            int tabs = StringUtil.countChars(lineTextForPosition, '\t');
+            int tabSize = getTabSize(editor);
+            int column = tabs * tabSize - tabs + lineTextForPosition.length();
+            int offset = editor.logicalPositionToOffset(new LogicalPosition(line, column));
+            if (pos.getCharacter() >= lineText.length()) {
+                LOG.debug(String.format("LSPPOS outofbounds: %s, line : %s, column : %d, offset : %d", pos,
+                        lineText, column, offset));
+            }
+            int docLength = doc.getTextLength();
+            if (offset > docLength) {
+                LOG.debug(String.format("Offset greater than text length : %d > %d", offset, docLength));
+            }
+            return Math.min(max(offset, 0), docLength);
+
         });
     }
 
     @Nullable
     public static LogicalPosition getTabsAwarePosition(Editor editor, Position pos) {
         return computableReadAction(() -> {
-            try {
-                if (editor.isDisposed()) {
-                    return null;
-                }
-                Document doc = editor.getDocument();
-                int line = Math.max(0, Math.min(pos.getLine(), doc.getLineCount()));
-                String lineText = doc.getText(DocumentUtil.getLineTextRange(doc, line));
-                String lineTextForPosition = !lineText.isEmpty() ? lineText.substring(0, min(lineText.length(),
-                        pos.getCharacter())) : "";
-                int tabs = StringUtil.countChars(lineTextForPosition, '\t');
-                int tabSize = editor.getSettings().getTabSize(editor.getProject());
-                int column = tabs * tabSize + lineTextForPosition.length() - tabs;
-                return new LogicalPosition(line, column);
-            } catch (IndexOutOfBoundsException e) {
+            if (editor.isDisposed()) {
                 return null;
             }
+            Document doc = editor.getDocument();
+            int line = max(0, Math.min(pos.getLine(), doc.getLineCount()-1));
+            String lineText = doc.getText(DocumentUtil.getLineTextRange(doc, line));
+            final int endCharInLine = max(0, min(lineText.length(), pos.getCharacter()));
+            String lineTextForPosition = endCharInLine > 0 ? lineText.substring(0, endCharInLine) : "";
+            int tabs = StringUtil.countChars(lineTextForPosition, '\t');
+            int tabSize = getTabSize(editor);
+            int column = tabs * tabSize + lineTextForPosition.length() - tabs;
+            return new LogicalPosition(line, column);
```

Comment:
@swissiety Sorry If I'm missing your point, but may I know the reason for removing the exception handling in here?

Reply:
`IndexOutOfBoundsException` was thrown because of using `doc.getLineCount()` as the last possible line - but the last line is doc.getLineCount()-1. Additionally negative values for the "char in line" have been possible wich made it possible for `substring(..)` to throw this exception. Now there shouldnt be a possibility to throw this exception.

Reply:
@swissiety got it.. Thanks!